### PR TITLE
Improve `merge_from_succ`

### PR DESCRIPTION
**Pull request by nnethercote**
A couple of small performance wins.

r? @nikomatsakis

### CODE REVIEW

**Comment by nnethercote** [comment]
Some local results:
```
clap-rs-check
        avg: -1.4%      min: -1.4%      max: -1.4%
inflate-check
        avg: -1.2%      min: -1.2%      max: -1.2%
cranelift-codegen-check
        avg: -0.6%      min: -0.6%      max: -0.6%
```

File: src/librustc_passes/liveness.rs [hunk 1]
```diff
@@ -843,6 +850,7 @@ impl<'a, 'tcx> Liveness<'a, 'tcx> {
 
             if changed {
                 this.rwu_table.assign_unpacked(idx, rwu);
+                any_changed = true;
```
Comment (nikomatsakis) [inline src/librustc_passes/liveness.rs]: I guess this code was just writing back changes far too often before? (i.e., it also triggered if any previous index had changed)

Comment (nikomatsakis) [reply src/librustc_passes/liveness.rs]: r=me if @nnethercote confirms this interpretation :)

Comment (nnethercote) [reply src/librustc_passes/liveness.rs]: Yes, as per the commit message (which unfortunately isn't visible via the "Files changed" tab).

**Comment by nnethercote** [comment]
Looks like the failure was an infra problem, unrelated to the commits.

@bors retry

**Pull request merged**